modules/styling-transform/lib/utils/handleCreateStencil.ts

```diff
@@ -62,7 +45,36 @@ export const handleCreateStencil: NodeTransformer = (node, context) => {
           property.name.text === 'extends' &&
           ts.isIdentifier(property.initializer)
         ) {
-          return getClassName(property.initializer.text, context);
+          const className = getClassName(property.initializer.text, context);
+          const extendsStencilName = className.split('-').slice(1).join('-');
+
+          if (
+            !Object.values(context.styles).reduce((result, fileStyles) => {
+              return (
+                result ||
+                fileStyles.reduce((_, rule) => {
+                  if (rule.includes(`.${className}`)) {
+                    return true;
+                  }
+                  return result;
+                }, false)
+              );
+            }, false)
```

Comment:
```suggestion
            !Object.values(context.styles).some(fileStyles => {
              return fileStyles.slice(-1).includes(`.${className}`);
            });
```
As it returns true only for last item that includes class name, I don't think it's good to use 2 reduce functions.

Reply:
You're right that the outer loop is a `.some`, which is more clear than a `.reduce`. But the `.slice(-1)` doesn't make any sense to me.

Both our code are wrong in different ways. The following is correct:

```ts
!Object.values(context.styles).some(fileStyles => {
  return fileStyles.some(rule => rule.includes(`.${className}`));
})
```